enhancements/proposals/1304-external-seeds/README.md

```diff
@@ -0,0 +1,156 @@
+This has proven to be incorrect in two ways: 
+1. Domain name addresses are also accepted as seeds. 
+2. There is no special significance to the first node of the first DC - it is just a simplification used for the purpose of minimising the risk of creating multiple clusters instead of a single, multi DC cluster.
+
+The specified constrains hold - the datacenters need a matching cluster name and distinct datacenter and rack names.
+
+### Risks and Mitigations
+
+#### Lookup of seed addresses on ScyllaDB initialization
+Currently, ScyllaDB performs an eager lookup of provided seeds on initialisation: https://github.com/scylladb/scylladb/blob/39ca07c49b257d37856164ea5b845f7b98224b98/init.cc#L35.
+An error in address lookup will hence cause ScyllaDB process to fail initialisation. This is expected during bootstrap.
+
+However, this is also going to cause an already bootstrapped node to fail after it has been restarted, e.g.:
+```
+ERROR 2023-08-03 08:35:54,663 [shard 0] init - Bad configuration: invalid value in 'seeds': 'dummy.scylla.svc.cluster.local': std::system_error (error C-Ares:4, dummy.scylla.svc.cluster.local: Not found)
+```
+
+Although the [documentation](https://opensource.docs.scylladb.com/stable/kb/seed-nodes.html) claims, that "**Once the nodes have joined the cluster, seed node has no function**", introducing the option to control the external seeds might cause instability to the ScyllaDB nodes controlled by the operator due to the above.
+To mitigate this, an issue has been opened with ScyllaDB core: [scylladb/scylladb#14945](https://github.com/scylladb/scylladb/issues/14945).
+
+## Design Details
+
+### ScyllaCluster API
+
+The following extension to the existing ScyllaCluster CR API is proposed:
+```go
+// ScyllaClusterSpec defines the desired state of Cluster.
+type ScyllaClusterSpec struct {
+    ...
+	
+	// externalSeeds specifies the external seeds to propagate to ScyllaDB binary on startup as "seeds".
+	// See https://opensource.docs.scylladb.com/stable/kb/seed-nodes.html.
+	// For the first running node in a datacenter, the external seeds are the only seeds provided.
+	// For the remaining nodes, one other node from their datacenter is appended to the external seeds.
+	// +optional
+	ExternalSeeds []string `json:"externalSeeds,omitempty"`
+}
+```
+
+The provided `externalSeeds` are passed to the sidecar through a new `--external-seeds` flag. StatefulSet reconciliation in ScyllaDB operator is modified accordingly.
+The sidecar's strategy for choosing seeds to provide to ScyllaDB processes is modified to follow the below pseudocode:
+```go
+func chooseSeeds(externalSeeds []string, seedFromLocalDC string) []string {
+  if is first node in DC {
+    if externalSeeds not empty {
+      return externalSeeds
+    }
+    return seedFromLocalDC
```

Comment:
if you are the first node, what's the seed here?

Reply:
added a comment about getting seeds from own DC

Reply:
who is the seedFromLocalDC if you are the first node (and the only node) in that DC?